datashader/core.py

```diff
@@ -282,25 +301,48 @@ def raster(self,
             lpct = lpad / (lpad + rpad) if lpad + rpad > 0 else 0
             left = int(np.ceil(num_width * lpct))
             right = num_width - left
-            left_pad = np.empty(shape=(self.plot_height, left)).astype(source_window.dtype) * np.nan
-            right_pad = np.empty(shape=(self.plot_height, right)).astype(source_window.dtype) * np.nan
+            lshape, rshape = (self.plot_height, left), (self.plot_height, right)
+            if bands > 1:
```

Comment:
I think we agreed to call these "layers" for generality, as "bands" mainly makes sense for satellite data, and channels was too confusing.